### DISCUSSION TITLE
Is oomph based setup of tycho develoment environment supported?

### DISCUSSION THREAD

**Discussion by carstenartur** [eclipse-tycho/tycho#2029] [Q&A] [answered]
<img width="427" alt="image" src="https://user-images.githubusercontent.com/3164220/214912732-4dfea0ff-7d18-4a46-8ecc-9fa5089f43a5.png">
It looks like it does not try to checkout from github. Dunno why...

**Comment by joeshannon** [comment]
The current setup file looks like it has the correct GitHub URL (changed in #142).
https://github.com/eclipse-tycho/tycho/blob/0710ad297d66936ff44035da631e0cd86fa95b26/setup/Tycho.setup#L143-L164
Is it somehow using an outdated Oomph index/file?

**Reply by carstenartur** [reply]
I used the update feature of the eclipse advanced installer. Is it not sufficient to proceed for setups from before the github migration?

<img width="1040" alt="image" src="https://user-images.githubusercontent.com/3164220/214931997-50380cbe-14cd-4ab1-8c96-63c3c71f87b4.png">

Maybe I should try to download the installer from scratch or remove the ".p2" folder?

**Comment by carstenartur** [comment, marked as answer]
Downloading a new installer obviously does not help
<img width="976" alt="image" src="https://user-images.githubusercontent.com/3164220/215327217-32f6d711-bc3d-47c9-8558-7044c7357b99.png">

There seem to be a lot of different urls involved in the resolution process eg.

https://git.eclipse.org/c/oomph/org.eclipse.oomph.git/plain/setups/org.eclipse.projects.setup
https://raw.githubusercontent.com/eclipse-tycho/tycho/master/setup/Tycho.setup#
http://www.eclipse.org/setups/setups.zip

But it is unclear how that works together and where the url for processing the git clone is coming from

<img width="962" alt="image" src="https://user-images.githubusercontent.com/3164220/215327782-b1ddf815-781b-4487-81fd-f93d3202898a.png">

**Reply by joeshannon** [reply]
Are there any references to the old clone URL in your Oomph user setup (located at `~/.eclipse/org.eclipse.oomph.setup/setups/user.setup` on Linux).

In mine I see:
```    
<setupTask
        xsi:type="setup:CompoundTask"
        name="Tycho">
      <restriction
          xsi:type="setup:Project"
          href="index:/org.eclipse.setup#//@projectCatalogs[name='org.eclipse']/@projects[name='tycho']"/>
      <setupTask
          xsi:type="setup:VariableTask"
          name="git.cloneTycho.remoteURI"
          value="https://github.com/eclipse/tycho.git"
          label="Tycho Github repository">
        <description>Choose from the available Github URIs</description>
      </setupTask>
```
I'm not entirely sure if this could cause issues, but possibly if there is a reference to the old URL in the user scope, or something similar, it could infer the incorrect clone URL as Oomph merges the scopes.

However this is just a guess and may be unrelated. Could also try running Oomph in ["isolated" mode](https://wiki.eclipse.org/Eclipse_Oomph_Authoring#Testing_a_Setup_in_a_Clean_Environment) and see what happens.

**Reply by HannesWell** [reply]
Besides this you could also check the ini-file of your eclipse Installation and remove any oomph.redirection for the tycho setup.
That could also cause the wrong Setup URL being used.

Or if you have provisioned a workspace to develop Oomph itself with the same Installation you should make Sure that the Oomph git clone is up to date.

**Reply by carstenartur** [reply]
I see the following content in the mentioned file

```xml
<setupTask
        xsi:type="setup:CompoundTask"
        name="Tycho">
      <restriction
          xsi:type="setup:Project"
          href="index:/org.eclipse.setup#//@projectCatalogs[name='org.eclipse']/@projects[name='tycho']"/>
      <setupTask
          xsi:type="setup:VariableTask"
          name="git.cloneTycho.remoteURI"
          value="https://git.eclipse.org/r/tycho/org.eclipse.tycho"
          label="Tycho Gerrit repository">
        <description>Choose from the available remote Gerrit URIs</description>
      </setupTask>
    </setupTask>
```

So it could be it interferes and and a fresh installation has no option to update these files?
I can safely delete it, right?
That is good enough for me at the moment.
Thanks!

**Reply by carstenartur** [reply]
@joeshannon I think just replacing the section you mentioned in ~/.eclipse/org.eclipse.oomph.setup/setups/user.setup or in my case
C:\Users\<username>\.eclipse\org.eclipse.oomph.setup\setups is sufficient

Before
```xml
<setupTask
        xsi:type="setup:CompoundTask"
        name="Tycho">
      <restriction
          xsi:type="setup:Project"
          href="index:/org.eclipse.setup#//@projectCatalogs[name='org.eclipse']/@projects[name='tycho']"/>
      <setupTask
          xsi:type="setup:VariableTask"
          name="git.cloneTycho.remoteURI"
          value="https://git.eclipse.org/r/tycho/org.eclipse.tycho"
          label="Tycho Gerrit repository">
        <description>Choose from the available remote Gerrit URIs</description>
      </setupTask>
    </setupTask>
```

After
```xml
<setupTask
        xsi:type="setup:CompoundTask"
        name="Tycho">
      <restriction
          xsi:type="setup:Project"
          href="index:/org.eclipse.setup#//@projectCatalogs[name='org.eclipse']/@projects[name='tycho']"/>
      <setupTask
          xsi:type="setup:VariableTask"
          name="git.cloneTycho.remoteURI"
          value="https://github.com/eclipse/tycho.git"
          label="Tycho Github repository">
        <description>Choose from the available Github URIs</description>
      </setupTask>
```

This seems to do the trick. I wonder if this would be possible to be done automatically.

**Comment by carstenartur** [comment]
[setup.log](https://github.com/eclipse-tycho/tycho/files/10529811/setup.log)
